src/auth/auth.service.ts

```diff
@@ -234,6 +243,124 @@ export class AuthService {
     }
   }
 
+  async forgotPasswordEmail(body: ForgotPasswordEmailDto) {
+    try {
+      const { email } = body;
+
+      const user = await this.prismaService.user.findUnique({ where: { email: email } });
+      if (!user) throw new NotFoundException(ErrorsTypes.NOT_FOUND_AUTH_USER);
+      const userId = user.id;
+      const token = await this.jwtService.signAsync(
+        { userId },
+        {
+          secret: this.config.get('FORGOT_PASSWORD_RESET_TOKEN_KEY'),
+          expiresIn: this.config.get('FORGOT_PASSWORD_RESET_TOKEN_EXPIRY'),
+        }
+      );
+
+      const hashLimit = this.config.get('MAX_HASH_LENGTH')!;
+
+      const hashedResetToken = await bcrypt.hash(
+        token.length > hashLimit ? token.slice(-hashLimit) : token,
+        parseInt(this.config.get('SALT_LENGTH')!)
+      );
+
+      await this.prismaService.user.update({
+        data: { passwordResetToken: hashedResetToken },
+        where: { id: user.id },
+      });
+
+      const forgotPasswordLink = new URL(
+        `/auth/forgot-password-reset?token=${token}`,
+        this.config.get('MAILER_CALLBACK_URL')
+      );
+
+      await this.mailerService.sendHtmlEmail(
+        email,
+        'Forgot password',
+        `<p> Please use this <a href="${forgotPasswordLink}">link</a> to set a new password. </p>`
+      );
+
+      return {
+        success: true,
+        message: translateMessage(this.i18n, MessagesTypes.AUTH_FORGOT_PASSWORD_EMAIL_SENT),
+      };
+    } catch (error) {
+      if (error instanceof HttpException) throw error;
+      throw new GlobalException(
+        ErrorsTypes.AUTH_FORGOT_PASSWORD_FAILED_TO_SEND_EMAIL,
+        error.message
+      );
+    }
+  }
+
+  async validateForgotPasswordToken(token: string) {
+    try {
+      const decodedToken = await this.jwtService.verify(token, {
+        secret: process.env.FORGOT_PASSWORD_RESET_TOKEN_KEY,
+      });
+
+      if (!decodedToken)
+        throw new BadRequestException(ErrorsTypes.FORBIDDEN_FORGOT_PASSWORD_INVALID_TOKEN);
+
+      const user = await this.prismaService.user.findUnique({ where: { id: decodedToken.userId } });
+
+      if (!user) throw new NotFoundException(ErrorsTypes.NOT_FOUND_AUTH_USER);
+
+      if (!user.passwordResetToken)
+        throw new BadRequestException(ErrorsTypes.FORBIDDEN_FORGOT_PASSWORD_INVALID_TOKEN);
+
+      const hashLimit = this.config.get('MAX_HASH_LENGTH')!;
+
+      if (token.length > hashLimit) {
```

Comment:
create please util to crop token, use it here and inside forgotPasswordEmail function. DRY. And make, please one update in this file. We use here somewhere such approach this.config.get('FORGOT_PASSWORD_RESET_TOKEN_KEY')
on another place it is 
secret: process.env.FORGOT_PASSWORD_RESET_TOKEN_KEY,
moreover, we getting it several times inside one file
so eject all values on the top of file like
const {SALT_LENGTH, FORGOT_PASSWORD_RESET_TOKEN_KEY ...} = process.env
const MaxHashLength = parseInt(MAX_HASH_LENGTH)
const SaltLength = parseInt(SALT_LENGTH)
...
and create technical task to eject all env values for each services. link this comment

Reply:
- created util func to crop tokens if too long;
- ejected env values for this service;
- created task for ejecting env values for other services, and have done it here;